src/main/java/edu/kit/kastel/sdq/intelligrade/state/PluginState.java

```diff
@@ -61,6 +63,35 @@ public static PluginState getInstance() {
         return pluginState;
     }
 
+    /**
+     * Logs out while displaying a warning if an assessment is still running
+     *
+     * @return true iff the logout was completed, false otherwise
+     */
+    public boolean logout() {
+        int answer = JOptionPane.OK_OPTION;
+        // check if confirmation is necessary because assessment is running
+        if (isAssessing()) {
+            answer = JOptionPane.showConfirmDialog(
+                    null,
+                    "Logging out while assessing will terminate the current assessment. Continue?",
```

Comment:
Not sure what you mean with "terminate", since the lock isn't freed. Maybe warn that local changes will be discarded, similar to the "Close Assessment" button?

Reply:
done